paasta_tools/bounce_lib.py

```diff
@@ -214,7 +213,7 @@ def kill_old_ids(old_ids, client):
             continue
 
 
-def get_happy_tasks(app, service, nerve_ns, min_task_uptime=None, check_haproxy=False):
+def get_happy_tasks(app, service, nerve_ns, system_paasta_config, min_task_uptime=None, check_haproxy=False):
```

Comment:
I originally had that, but decided against it. `get_happy_tasks` already does a lot of things (looks at marathon, synapse, etc.), and conceivably could do more in the future. If it ever needs to look at some other system config parameter, I'd rather not add yet another argument.

The counterargument would be: What if the synapse port that `get_happy_tasks` looked at depended on context? (e.g. we ran multiple different synapses for different purposes or something). I'm not sure I ever see that happening.

I'm starting to think that our convention for system paasta configs would look like:

 - Main functions or other "entry points", like `deploy_service`, would call `load_system_paasta_config`.
 - Intermediate functions (like `get_happy_tasks`) would generally have `system_paasta_config` as an argument
 - Single-purpose utility functions (like `get_registered_marathon_tasks`) would not take `system_paasta_config`, and instead would take specific parameters, like `synapse_port`.